### ISSUE TITLE
Train on new dataset

### ISSUE THREAD

**Issue by harlanhong** [HumamAlwassel/TSP#1]
How can I train on a new dataset? just follow the way of data preprocessing in ''TSP/data'' to prepare my new dataset?

**Comment by HumamAlwassel** [comment]
Hi Harlan,

Thanks for your interest in our work.

To train on a new localization dataset, you need the following:
- Preprocess the videos following the instructions in [TSP/data](https://github.com/HumamAlwassel/TSP/tree/master/data).
- Train and valid groundtruth CSV files with annotation for the two heads: `action-label` and `temporal-region-label`. Find an example in [TSP/data/activitynet/activitynet_v1-3_train_tsp_groundtruth.csv](https://github.com/HumamAlwassel/TSP/blob/master/data/activitynet/activitynet_v1-3_train_tsp_groundtruth.csv). These CSV files are built using the temporal action annotations of ActivityNet and THUMOS14. Thus, you will need to do the same for the new dataset based on its temporal annotations. 
- Action label mapping JSON file to specify the names of the actions in the dataset. Find an example in [TSP/data/activitynet/activitynet_v1-3_action_label_mapping.json](https://github.com/HumamAlwassel/TSP/blob/master/data/activitynet/activitynet_v1-3_action_label_mapping.json)
- Finally, you need to edit the input data parameters (groundtruth CSV + label mapping JSON files) in the training scripts to point to the ones of your new dataset.

Cheers,
Humam

**Comment by satish1901** [comment]
Hi Humam
How are you doing ?
I have a quick follow up question on training on new dataset. 
How do i generate the GVF features to train on my data?
I see its an optional argument but if i want to use it then how?
